pkg/builder/builder.go

```diff
@@ -567,3 +616,80 @@ func HasProhibited(tags []string) bool {
 
 	return false
 }
+
+func ContainsActions(target string, sas []schemas.ScheduledAction) bool {
+	for _, sa := range sas {
+		if sa.Name == target {
+			return true
+		}
+	}
+	return false
+}
+
+// ValidCronExpression checks if the cron expression is valid or not
+// It should be [Minute] [Hour] [Day_of_Month] [Month_of_Year] [Day_of_Week]
+func ValidCronExpression(expression string) (bool, error) {
+	elements := strings.Split(expression, " ")
+	if len(elements) != 5 {
+		return false, fmt.Errorf("cron expression should be in the format of [Minute] [Hour] [Day_of_Month] [Month_of_Year] [Day_of_Week]: %s", expression)
+	}
+
+	// minutes
+	if len(elements[0]) > 0 && elements[0] != "*" {
+		i, _ := strconv.Atoi(elements[0])
+		if i < 0 || i > 60 {
+			return false, fmt.Errorf("first element should be from 0 to 60 or *: %s", expression)
+		}
+	}
+
+	// hours
+	if len(elements[1]) > 0 && elements[1] != "*" {
+		hours := strings.Split(elements[1], ",")
+		for _, h := range hours {
+			i, _ := strconv.Atoi(h)
+			if i < 0 || i > 24 {
+				return false, fmt.Errorf("second element should be from 0 to 24 or *: %s", expression)
```

Comment:
May be it would be 0 to 23, isn't it?

Reply:
I fixed them all based on the information you mentioned. Thanks :)

Reply:
Thank you!